src/cvc.ml

```diff
@@ -0,0 +1,295 @@
+let try_with_asn f = try Result.Ok (f ()) with Asn.Parse_error s -> Result.Error s
+let raise_asn f = match f () with Result.Ok x -> x | Result.Error s -> Asn.parse_error s
+
+let pp_of_to_string to_string fmt x =
+  Format.pp_print_string fmt (to_string x)
+
+module Asn = struct
+  include (Asn : module type of Asn with module OID := Asn.OID and type 'a t = 'a Asn.t)
+
+  module OID = struct
+    include Asn.OID
+    let pp = pp_of_to_string to_string
+    let compare a b =
+      String.compare (to_string a) (to_string b)
+
+    let of_yojson = function
+      | `String s -> Result.Ok (Asn.OID.of_string s)
+      | _ -> Result.Error "Cannot convert this json value to Asn.OID.t"
+
+    let to_yojson oid =
+      `String (Asn.OID.to_string oid)
+  end
+end
+
+module Z = struct
+  include Z
+  let pp = pp_of_to_string to_string
+
+  let of_yojson = function
+    | `String s -> Result.Ok (Z.of_string s)
+    | _ -> Result.Error "Cannot convert this json value to Z.t"
+
+  let to_yojson z =
+    `String (Z.to_string z)
+end
+module RSA =
+struct
+  module Public =
+  struct
+    type t = {
+      n: Z.t;
+      e: Z.t;
+    }
+    [@@deriving ord,yojson,eq,show]
+  end
+end
+
+module ECDSA =
+struct
+  module Public =
+  struct
+    type t =
+      { modulus : Z.t
+      ; coefficient_a : Z.t
+      ; coefficient_b : Z.t
+      ; base_point_g : Z.t
+      ; base_point_r_order : Z.t
+      ; public_point_y : Z.t
+      ; cofactor_f : Z.t
+      }
+      [@@deriving ord,yojson,eq,show]
+  end
+end
+
+let rsa_oid = Asn.OID.of_string "0.4.0.127.0.7.2.2.2.1.4"
+let ecdsa_oid = Asn.OID.of_string "0.4.0.127.0.7.2.2.2.2.3"
+
+type t =
+  [ `RSA of RSA.Public.t | `ECDSA of ECDSA.Public.t | `UNKNOWN ]
+
+type algo_typ =
+  | RSA
+  | ECDSA
+  | Unknown of Asn.OID.t
+
+type parser_state =
+  | Init
+  | Type
+  | Length
+  | Value of int
+
+let cvc_object_types =
```

Comment:
Is it particularly harmful to keep them around in case we'd like to add them later? I think it's also nice to avoid jumps in the numbering.